sdk/python/kfp/dsl/_for_loop.py

```diff
@@ -48,10 +51,21 @@ def __init__(self, items: ItemList, code: Text):
                                      "name.".format(subvar_name))
                 setattr(self, subvar_name, LoopArgumentVariable(self.name, subvar_name))
 
-        self.items = items
+        self.items_or_pipeline_param = items
+        self.referenced_subvar_names = []
+
+    def __getattr__(self, item):
+        # this is being overridden so that we can access subvariables of the LoopArguments (i.e.: item.a) without
+        # knowing the subvariable names ahead of time
+        self.referenced_subvar_names.append(item)
+        return LoopArgumentVariable(self.name, item)
 
     def to_list_for_task_yaml(self):
-        return self.items
+        if isinstance(self.items_or_pipeline_param, list):
+            return self.items_or_pipeline_param
+        else:
+            raise ValueError("You should only call this method on loop args which have list items, "
+                             "not pipeline param items.")
```

Comment:
I can't seem to be able to handle tuples, and I'm getting this error message. Maybe these changes broke it?

Reply:
really I was only shooting for list inputs, but i've added an option which will make it more lenient to tuples.